lib/moonshot/build_mechanism/github_release.rb

```diff
@@ -170,6 +169,27 @@ def fetch_changes(version)
       end
     end
 
+    # Checks for the commit's CI job status. If its not finished yet,
+    # wait till timeout.
+    #
+    # @param sha [String] Commit sha.
+    #
+    # @return [String] Status and links to the CI jobs
+    def check_ci_status(sha)
+      out = nil
+      ilog.start_threaded('Validate `CHANGELOG.md`.') do |step|
+        Retriable.retriable on: RuntimeError,
```

Comment:
There are not many common values to share. Should find a nice location to put them in. I errd on defining them locally cos evry retry is kinda different in its own way.
But, I was thinking adding a retry option to [sh_out](https://github.com/kaushik/moonshot/blob/CPD-1834/lib/moonshot/shell.rb#L7) might be a good choice.
`
def sh_out(cmd, fail: true, stdin: '', retry: false, opts = {})
default_retry_optons.merge!(opts)
.
`